pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -166,7 +166,14 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	if mapCtx.Err() != nil {
 		return mapCtx.Err()
 	}
-	status.ExternalRef = &created.Name
+
+	tokens := strings.Split(created.Name, "/")
+	parent, err = a.id.Parent()
```

Comment:
Idea for future: I'm wondering if we should introduce a type ProjectLocationRef.  It wouldn't be used by any API (at least I haven't seen one yet!), but it might make Parent() easier.